### ISSUE TITLE
Quick thoughts after skimming the dataa manually

### ISSUE THREAD

**Issue by msaroufim** [PaliC/categorize_triton_functions#1]
- Fun facts

- Activation functions
  - Mostly have variants of relu, swilu and swiglu - these are all generatable via torch.compile?

- Embedding
  - This is mostly rotary embedding kernel. I believe gpt-fast has figured out how to compile this already. Embedding code is also frankly hard to read

- Attention mechanism
  - Most diverse set so far - flex attention might be doing good stuff here

- Gradient computation
  - This label feels misleading it's closer to backwards kernels, this probably shouldn't be a category in of itself BUT might be OK in the case when training ends up being a tiny fraction of usage of triton online. Sorta makes sense considering in torch people are using to getting autograd for free and that's not the case for triton

- Grid sampling
  - Don't have a good sense of what this used for quite yet

- Layer norm
  - The most similar and was my theory of what might be the most common kernel. In particular here I'd be curious about if the kernels are identical or if they use different performance tricks

- Loss functions
  - AS expected cross entropy is super popular

- Matrix operations
  - Matrix operations for gemm and gemv with variants for quantization

- Memory management
  - Lots of kv cache management stuff

- Utility
  - Not a wel named category but includes individual functions to things like bit packing, seems to be an op centric view
